tools/DevDiagnostics/DevHome.DevDiagnostics/Models/SensorNode.cs

```diff
@@ -0,0 +1,294 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using CommunityToolkit.Mvvm.ComponentModel;
+using DevHome.DevDiagnostics.Helpers;
+using LibreHardwareMonitor.Hardware;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Windows.AI.MachineLearning;
+
+namespace DevHome.DevDiagnostics.Models;
+
+public partial class SensorNode : ObservableObject, IHardwareMonitorTreeNode
+{
+    private readonly ISensor _sensor;
+    private readonly string _unitFormat;
+
+    public SensorNode(ISensor s)
+    {
+        _name = s.Name;
+        _value = GetValueString();
+        _sensor = s;
+
+        // Set the default format to "{0}" if no specific format is found
+        _unitFormat = "{0}";
+
+        switch (_sensor.SensorType)
+        {
+            case SensorType.Voltage:
+                _unitFormat = "VoltageSensorFormat";

```

Comment:
I wonder... if this is a fixed relationship, might it make sense to define a map of some kind?

Reply:
I'll keep the switch statement for now since this mapping is only used once and this way we don't need to create and populate data for a dictionary.